h/static/scripts/tests/controllers/search-bar-controller-test.js

```diff
@@ -221,30 +231,37 @@ describe('SearchBarController', function () {
       });
   });
 
-  it('allows submitting the form when query is empty and no dropdown element is selected', function (done) {
+  it('does not submit the form when a dropdown element is selected', function (done) {
     let submitted = false;
-    form.addEventListener('submit', event => {
+    form.addEventListener('submit', () => {
       submitted = true;
     });
 
     syn
       .click(input)
-      .type('[enter]', () => {
-        assert.isTrue(submitted);
+      .type('[down][enter]', () => {
+        assert.isFalse(submitted);
         done();
       });
   });
 
-  it('does not submit the form when a dropdown element is selected', function (done) {
-    let submitted = false;
-    form.addEventListener('submit', event => {
-      submitted = true;
-    });
+  it('allows submitting the form when query is empty and no dropdown element is selected', function (done) {
+    var submit = sinon.stub(form, 'submit');
 
     syn
       .click(input)
-      .type('[down][enter]', () => {
-        assert.isFalse(submitted);
+      .type('[enter]', () => {
+        assert.isTrue(submit.calledOnce);
+        done();
+      });
+  });
+
+  it('should create a lozenge when the user presses space', function (done) {
+    syn
+      .click(input)
+      .type('bar')
+      .type('[space]', () => {
+        assert.equal(testEl.querySelectorAll('.js-lozenge__content')[1].textContent, 'bar');
```

Comment:
I think a few more new tests for `search-bar-controller.js` are needed, maybe:

* Test that when there is text in the hidden input when the controller is instantiated, then it lozengifies that text and empties the hidden input. Test with various kinds of texts in the hidden input (single lozenge, multiple lozenges, facets, quotes...). For example there is a bug that if I load this URL: <http://localhost:5000/search?q=%22foo+bar>, which contains an unclosed search term, then a lozenge for that search term is created, and weird things happen from there, for example if I just keep submitting the form by pressing `Enter` then it keeps adding `+`s to the query:

![peek 2016-10-18 19-18](https://cloud.githubusercontent.com/assets/22498/19490644/e3024b62-9567-11e6-9824-83aa8c5bcd28.gif)

Or if I type something into the search and either lozengify it or don't, on `Enter` it'll get added to the first, unclosed lozenge:

![peek 2016-10-18 19-21](https://cloud.githubusercontent.com/assets/22498/19490730/2af7b074-9568-11e6-932a-ccd188d382ff.gif)

![peek 2016-10-18 19-22](https://cloud.githubusercontent.com/assets/22498/19490734/2f1d5848-9568-11e6-9f47-27450af3a479.gif)

* Test what happens when the form is submitted (updating the hidden input value from the lozenges then submitting the form). Test with various states: no lozenges, one lozenge, multiple lozenges, no unlozengified text in the input, some unlozengified text in the input, the text in the input is a single word or is a quoted multi-word or is a quoted multi-word with no closing quote.

   For example, it's possible to get into the "unclosed lozenge" situation above like this: Type `"foo` into the search but don't type the closing quote and don't press `Space`. Then press `Enter`. Now you have an unclosed `"foo` lozenge that will "eat" any other lozenges on `Enter`:

![peek 2016-10-18 19-26](https://cloud.githubusercontent.com/assets/22498/19490934/08d60f4e-9569-11e6-902a-9ce5f816d5be.gif)

* I'd add more tests for creating lozenges when `Space` is pressed, e.g. if there's an unclosed quote it should _not_ create one.